generate/generate.go

```diff
@@ -1339,21 +1339,23 @@ func loadStackCodeCfgs(
 	return genfilesConfigs, nil
 }
 
-func cleanupOrphaned(root *config.Root, report Report) Report {
+func cleanupOrphaned(root *config.Root, target *config.Tree, report Report) Report {
 	logger := log.With().
 		Str("action", "generate.cleanupOrphaned()").
+		Stringer("dir", target.Dir()).
 		Logger()
-	// If the root of the tree is a stack then there is nothing to do
-	// since there can't be any orphans (the root parent stack owns
-	// the entire project).
-	if root.Tree().IsStack() {
-		logger.Debug().Msg("project root is a stack, nothing to do")
+
+	defer report.sort()
```

Comment:
by luck I got a failing test in the macos CI runner. The problem was in the early return:
```
if target.IsStack() {
	return report
}
```
which was not sorting the report in the case root was an stack.
This only triggers in the new test I added, and rarely for some reason..